external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MySQLIntegrationSuite.scala

```diff
@@ -127,9 +127,10 @@ class MySQLIntegrationSuite extends DockerJDBCIntegrationSuite with V2JDBCTest {
     val properties = new util.HashMap[String, String]();
     properties.put("KEY_BLOCK_SIZE", "10")
     properties.put("COMMENT", "'this is a comment'")
+    properties.put(SupportsIndex.INDEX_TYPE, "BTREE")
     // MySQL doesn't allow property set on individual column, so use empty Array for
     // column properties
-    jdbcTable.createIndex("i1", "BTREE", Array(FieldReference("col1")),
+    jdbcTable.createIndex("i1", Array(FieldReference("col1")),
```

Comment:
TBH it's a bit weird to write UT in this suite. Can we turn it into end-to-end test now as we have the SQL syntax ready?